packages/react/ds/styles.css

```diff
@@ -10,3 +10,17 @@
   max-height: 0;
   opacity: 0;
 }
+
+/* Header Styles */
+
+#MenuContainer:has(#SearchTrigger:checked) ~ #SearchContainer {
```

Comment:
I'm not sure about this.
You should create a class used only for the container instead of using the IDs.
Check this guide
https://tailwindcss.com/docs/adding-custom-styles#adding-component-classes

Or is there something that I'm missing?

Reply:
what I'm trying to do here ( styling an element based on another element's state )  can be achieved through the `peer` class 
https://tailwindcss.com/docs/hover-focus-and-other-states#styling-based-on-sibling-state

I've tried that approach but for some reason the peer class does not work as intended.